src/tree/hist/evaluate_splits.h

```diff
@@ -367,8 +367,9 @@ class HistEvaluator {
     auto const &cut_ptrs = cut.Ptrs();
     // Under secure vertical setting, only the active party is able to evaluate the split
     // based on global histogram. Other parties will receive the final best split information
-    // Hence the below computation is not performed by the passive parties
-    if ((!is_secure_) || (collective::GetRank() == 0)) {
+    // Hence the below computation is not performed by the passive parties for secure vertical
+    // All other cases need it
+    if (not (is_col_split_ && is_secure_ && collective::GetRank() != 0)) {
```

Comment:
We should have a single function is collective to query this type of information.